yaksh/forms.py

```diff
@@ -274,6 +274,17 @@ def __init__(self, *args, **kwargs):
 class CourseForm(forms.ModelForm):
     """ course form for moderators """
 
+    def save(self, commit=True, *args, **kwargs):
```

Comment:
I think having two fields i.e. **code** and **hidden** for this feature is slightly increasing the complexity.
I mean hidden field is of no use for the moderator. **hidden** field is not changeable by the moderator, it simply depends on the course code. So each time we need to handle hidden field as per the course code. 

We can simply have a **code** field, that solely determines everything for this feature.
If you want to display only those courses that do not have code then the following can be done:
```Course.objects.exclude(code__isnull=True)```
I think this will reduce lot of your lines of code and makes it simpler.